src/main/java/projet/projetstage02/service/GestionnaireService.java

```diff
@@ -685,4 +687,11 @@ public StageContractOutDTO contractSignature(SignatureInDTO signature)
 
         return new StageContractOutDTO(contract);
     }
+
+    public void sendNewOfferEmail(long offerId) throws NonExistentOfferExeption {
+        Offre offer = getOfferById(offerId);
+        List<String> emails =
+                studentRepository.findAll().stream().filter(AbstractUser::isConfirm).map(Student::getEmail).toList();
```

Comment:
Only send this email to the students of the department. I mean I don't want to get a notification of Transport, if I am in Informatique.